src/scripts/farming/Farming.ts

```diff
@@ -5,63 +5,894 @@ class Farming implements Feature {
+                {berryType: BerryType.Nanab, berryStage: PlotStage.Berry},
+                {berryType: BerryType.Magost, berryStage: PlotStage.Berry},
+                {berryType: BerryType.Watmel, berryStage: PlotStage.Berry},
+            ]));
+        //#endregion
+
+        //#region Fifth Generation
+
+        // Micle
+
+        // Custap
+
+        // Jaboca
+
+        // Rowap
+
+        // Kee
+
+        // Maranga
+
+        // Liechi
+
+        // Ganlon
+
+        // Salac
+
+        // Petaya
+
+        // Apicot
+
+        // Lansat
+
+        // Starf
+
+        //Enigma
+
+
+        //#endregion
+
+        //#endregion
+
+    }
+
+    getGrowthMultiplier(): number {
+        let multiplier = 1;
+        multiplier *= App.game.oakItems.calculateBonus(OakItems.OakItem.Sprayduck);
+        return multiplier;
+    }
+
+    getReplantMultiplier(): number {
+        let multiplier = 1;
+        multiplier *= App.game.oakItems.calculateBonus(OakItems.OakItem.Squirtbottle);
+        return multiplier;
+    }
+
+    getMutationMultiplier(): number {
+        let multiplier = 1;
+        multiplier *= App.game.oakItems.calculateBonus(OakItems.OakItem.Sprinklotad);
+        return multiplier;
     }
 
     update(delta: number): void {
-        const timeToReduce = delta * App.game.oakItems.calculateBonus(OakItems.OakItem.Sprayduck);
-        let notify = false;
+        const timeToReduce = delta;
+
+        const notifications = new Set<FarmNotificationType>();
+
+        let change = false;
+
         this.plotList.forEach(plot => {
-            plot.reduceTime(timeToReduce);
-            if (!plot.isEmpty() && plot.timeLeft == 0 && !plot.notified) {
-                plot.notified = true;
-                notify = true;
+            if (plot.update(timeToReduce)) {
+                change = true;
+            }
+            if (plot.notifications) {
+                plot.notifications.forEach(n => notifications.add(n));
+                plot.notifications = [];
             }
         });
-        if (notify) {
-            Notifier.notify({
-                message: 'A berry is ready to harvest!',
-                type: NotificationConstants.NotificationOption.success,
-                sound: NotificationConstants.NotificationSound.ready_to_harvest,
-                setting: NotificationConstants.NotificationSetting.ready_to_harvest,
+
+        this.counter += GameConstants.TICK_TIME;
+        if (this.counter >= GameConstants.MUTATION_TICK) {
+            this.mutations.forEach(mutation => {
+                if (mutation.mutate()) {
+                    notifications.add(FarmNotificationType.Mutated);
+                    change = true;
+                }
             });
+            this.counter = 0;
+        }
+
+        if (change) {
+            this.resetAuras();
+        }
+
+        if (notifications.size) {
+            notifications.forEach(function(n) {
+                this.handleNotification(n);
+            }, this);
```

Comment:
I haven't tried it, but this can probably be changed to
```js
notifications.forEach(this.handleNotification)
```

Reply:
same here